WireDomain/Sources/WireDomain/Repositories/Conversations/LocalStore/ConversationLocalStore.swift

```diff
@@ -1236,6 +1236,37 @@ public final class ConversationLocalStore: ConversationLocalStoreProtocol {
         }
     }
 
+    // MARK: - Backup / Restore
+
+    public func totalConversationCountForBackup() async throws -> Int {
+        try await context.perform { [context] in
+            try context.count(for: ZMConversation.fetchRequestForBackup())
+        }
+    }
+
+    public func fetchAllConversationIDsForBackup() async throws -> [QualifiedID] {
+        let fetchRequest = ZMConversation.fetchRequestForBackup()
+        fetchRequest.propertiesToFetch = ["remoteIdentifier_data", "domain"]
+        return try await context.perform { [context] in
+            let conversations = try context.fetch(fetchRequest)
+            return conversations.compactMap(\.qualifiedID)
+        }
+    }
+
+    public func fetchAllConversationsForBackup() async throws -> [ZMConversation] {
+        try await context.perform { [context] in
+            try context.fetch(ZMConversation.fetchRequestForBackup())
+        }
+    }
```

Comment:
suggestion: as I have mentioned in a previous comment, I think it may be worth to have a specific backup store. I've discussed in the past how these entity based stores are growing and growing based on different needs and it's getting a bit messy. See for instance https://github.com/wireapp/wire-ios/blob/develop/wire-ios-sync-engine/Source/SessionManager/UpdateEventMigratorDAO.swift

Reply:
Done, created a BackupLocalStoreProtocol as you suggested.